compiler/rustc_mir_build/src/build/custom/mod.rs

```diff
@@ -118,7 +118,7 @@ fn parse_attribute(attr: &Attribute) -> MirPhase {
                 phase = Some(value);
             }
             other => {
-                panic!("Unexpected key {}", other);
+                bug!("Unexpected key while parsing custom_mir attribute: '{}'", other);
```

Comment:
```suggestion
                span_bug!(nested.span, "Unexpected key while parsing custom_mir attribute: '{}'", other);
```

I think this works?

Reply:
ah, neat! :)
![mir_ice](https://github.com/rust-lang/rust/assets/476013/72ce1388-0983-4b8d-a4dd-d7cbd9e29ea2)